### PR TITLE
feat/add dao financial statements

### PR DESCRIPTION
**Pull request by hetfly**

### CODE REVIEW

**Comment by hetfly** [comment]
@parrot9design 
statements
There are 2 columns each representing single month. Selected time period says past month. Is this intended? What happens when I change the time period? Will there always be 2 data columns ?

**Comment by parrot9design** [comment]
> There are 2 columns each representing single month. Selected time period says past month. Is this intended? What happens when I change the time period? Will there always be 2 data columns ?

Yeah this didn’t make sense to us either, until Scott broke it down. 😅 

It uses a trailing time period, so when you change the time period, both columns will update to show the correct date ranges and data. The first column will always show data for the time period ending today, and the second column shows the previous period. 

So if you selected `past month` today, the first column would show financial data for the month ending today (AUG 3 - SEP 2), and the second would show data for the previous month (JUL 3 - AUG 2).

If you were to change it to `past week`, the columns would show AUG 27 - SEP 2 and AUG 20 -AUG 26.  `past day` would show SEP 1 - SEP 2 and AUG 31 - SEP 1 (unless keeping them to single days – SEP 2 and SEP 1 – would be more accurate for reporting @bennyoptions).

We’ll always have two columns so it’s easy for people to compare the financials between the current and previous period.

As a side note, I noticed we have the $ value of Balanced Dollars under `Loan fees`. I’m not sure how that made it into the design we shared with you, but let’s remove that and only show the number of tokens, like under `Swap fees`.

**Comment by hetfly** [comment]
Makes sense now, thanks @parrot9design 
So half a year and a year periods will be available after Balanced being live for 2x the time period ?
Are APIs ready or should I contact someone about it ?

**Comment by parrot9design** [comment]
Yep, that’s correct! `past 6 months` and `past year` can be included when we get to the 1 year and 2 year mark, respectively. So `past 3 months` is the highest time frame we can include for now (maybe even 2 months, considering we’re only just past 4 months of operation). 

I think the Geometry team are still working on the APIs, but @BennyOptions will have more insight into that.

**Comment by hetfly** [comment]
Waiting for the APIs

**Comment by hetfly** [comment]
```Holdings``` are static.
Values in ```Earnings``` starting with ```**``` are static and are missing in API response
I'm aware of some values displaying as ```0``` right after you reload the page

@parrot9design can you please check the numbers for loan fees ?

**Comment by parrot9design** [comment]
For the `Earnings` dropdown, let's use these options:

- day
- week
- month
- 2 months

---
For the Loan fees, let's remove the $ values from the bnUSD row, so each column only displays a single number.

**Current** | **Ideal**
----------- | ---------
Screen Shot 2021-09-20 at 3 17 03 PM | Screen Shot 2021-09-20 at 3 17 49 PM

Also, remove the comma from the date formatting (shown in the images above, as well).

---

>Holdings are static.
>
>Values in Earnings starting with ** are static and are missing in API response

To clarify, only the Loan fee API is working at this stage, then? Do you need us to provide a date selector design for the `Holdings` section, or are you okay to try this on your own?

**Comment by hetfly** [comment]
> To clarify, only the Loan fee API is working at this stage, then? Do you need us to provide a date selector design for the `Holdings` section, or are you okay to try this on your own?

Yes, only loan fees are working right now. I also have data for the ```Holdings```  current day but not for any past date.
I've contacted Pranav about the API issues.

There is no need for date picker design

**Comment by hetfly** [comment]
@parrot9design  time picker added

**Comment by pranav925** [comment]
The daofund balance sheet endpoint is up here: https://balanced.geometry.io/api/v1/stats/daofund-balance-sheet?timestamp=-1

You can test out different parameters with this page: https://balanced.geometry.io/api/v1/docs#/stats/get_daofund_balance_sheet_api_v1_stats_daofund_balance_sheet_get

NOTE: the timestamps are unix timestamps in microseconds. A `-1` timestamp will give you the latest balance sheet
Example query to get daofund balance sheet of the last week: https://balanced.geometry.io/api/v1/stats/daofund-balance-sheet?timestamp=1631575246128000

This endpoint will only get you the bottom half of the page, the income statement endpoint are a WIP. Looking to get it finished by EOW.

**Comment by BennyOptions** [comment]
@hetfly all numbers have been verified within reason. 

@octocat512 if you want to review the code prior to merging, please do.

Let's merge this PR on Monday or Tuesday night US time when we announce the new roadmap.

**Comment by hetfly** [comment]
@octocat512 @BennyOptions 
I want to do some refactoring, will do so prior to Monday

File: src/pages/PerformanceDetails/sections/EarningSection/index.tsx [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+            {earningsPastPeriod ? (
+              displayValueOrLoader(loanFeesPast, 1)
+            ) : (
+              <StyledSkeleton animation="wave" width={100} />
+            )}
+          </GridItemStrong>
+        </IncomeGrid>
+
+        <IncomeGrid>
+          <GridItemLight>Balanced Dollars (bnUSD)</GridItemLight>
```
Comment (parrot9design) [inline src/pages/PerformanceDetails/sections/EarningSection/index.tsx:200]:
```suggestion
          <GridItemLight>Balanced Dollar (bnUSD)</GridItemLight>
```

I always prefer to keep the `s` on Balanced Dollars, Balance Tokens, and Omm Tokens, but let's remove it here to keep it consistent with the rest of the Stats page.

Comment (hetfly) [reply src/pages/PerformanceDetails/sections/EarningSection/index.tsx]: fixed

**Review by parrot9design — CHANGES_REQUESTED** [review]
Apologies for missing this earlier, but I noticed the asset tickers are a bit messy in the Earnings section:
Screen Shot 2021-10-05 at 12 52 04 PM
 
Let's only display the asset tickers next to the amount in the date columns. That means we need to add `bnUSD` to the amounts for `Loan fees`, and remove the tickers from the fee line items:

Screen Shot 2021-10-05 at 1 04 54 PM

File: src/pages/PerformanceDetails/index.tsx [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+      <Breadcrumbs>
+        <BreadcrumbsLink to="/">Stats</BreadcrumbsLink>
+        <BreadcrumbsSeparator>{'>'}</BreadcrumbsSeparator>
+        <BreadcrumbsCurrent>Performance details</BreadcrumbsCurrent>
+      </Breadcrumbs>
+
+      <Typography fontWeight="bold" fontSize={[45, 45, 60]} color="#fff">
+        Performance details
+      </Typography>
+      <Perex>
```
Comment (octocat512) [inline src/pages/PerformanceDetails/index.tsx:167]: The Typography component isn't fit for this use case?

File: src/pages/PerformanceDetails/utils/index.tsx [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+    animation-delay: 400ms;
+  }
+`;
+
+export const getTimestampFrom = (from: number = 0): number => {
+  const date = new Date();
+  return date.setDate(date.getDate() - from) * 1_000;
+};
+
+export const displayValueOrLoader = (
```
Comment (octocat512) [inline src/pages/PerformanceDetails/utils/index.tsx:132]: I'd like you to make this function a React component.

**Review by octocat512 — CHANGES_REQUESTED** [review]

**Comment by hetfly** [comment]
@octocat512  @parrot9design 
thanks for your feedback, everything is fixed

**Review by parrot9design — APPROVED** [review]
Looks great, thanks @hetfly! 🙏

**Pull request merged**